pss/pss.go

```diff
@@ -52,11 +61,37 @@ func (p *Pss) Send(ctx context.Context, targets [][]byte, topic trojan.Topic, pa
 		return nil, err
 	}
 
+	tag, err := p.tags.Create("pss-chunks-tag", 1, false)
+	if err != nil {
+		return nil, err
+	}
+
 	// SAVE trojanChunk to localstore, if it exists do nothing as it's already peristed
-	// TODO: for second phase, use tags --> listen for response of recipient, recipient offline
-	if _, err = p.localStore.Put(ctx, chunk.ModePutUpload, tc); err != nil {
+	if _, err = p.localStore.Put(ctx, chunk.ModePutUpload, tc.WithTagID(tag.Uid)); err != nil {
 		return nil, err
 	}
+	tag.Inc(chunk.StateStored)
+	tag.Total = 1
 
-	return tc, nil
+	monitor := &Monitor{
+		chunk: tc,
+		tag:   tag,
+		state: make(chan chunk.State, 3),
+	}
+
+	go monitor.updateState()
+
+	return monitor, nil
+}
+
+func (m *Monitor) updateState() {
+	for _, state := range []chunk.State{chunk.StateStored, chunk.StateSent, chunk.StateSynced} {
+		for {
+			n, total, err := m.tag.Status(state)
+			if err == nil && n == total {
+				m.state <- state
+				break
+			}
```

Comment:
this needs to be Sleep here! otherwise you laptop melts if the network doesn't work

Reply:
I left a 1-second delay, but what would be a recommended interval?

Reply:
10ms or 100ms is better

Reply:
fixed